packages/electron/src/renderer/addBacktraceElectron.ts

```diff
@@ -0,0 +1,53 @@
+import { BacktraceCoreClientBuilder } from '@backtrace-labs/sdk-core';
+import { IpcEvents } from '../common/ipc/IpcEvents';
+import { SyncData } from '../common/models/SyncData';
+import { getIpcRpc } from './ipc/getIpcRpc';
+import { getIpcTransport } from './ipc/getIpcTransport';
+import { ConstSessionProvider } from './modules/ConstSessionProvider';
+import { IpcAsyncSessionProvider } from './modules/IpcAsyncSessionProvider';
+import { IpcBreadcrumbsStorage } from './modules/IpcBreadcrumbsStorage';
+import { IpcReportSubmission } from './modules/IpcReportSubmission';
+import { IpcRequestHandler } from './modules/IpcRequestHandler';
+import { IpcSummedMetricsQueue } from './modules/IpcSummedMetricsQueue';
+import { StubMetricsQueue } from './modules/StubMetricsQueue';
+
+export interface AddBacktraceElectronOptions {
+    readonly synchronous?: boolean;
+}
+
+export function addBacktraceElectron<T extends BacktraceCoreClientBuilder>(
+    builder: T,
+    options?: AddBacktraceElectronOptions,
+): T {
+    const ipcTransport = getIpcTransport();
+    const ipcRpc = getIpcRpc();
+
+    // Sanity check - this will throw if ping isn't listened on.
+    ipcRpc
+        .invoke(IpcEvents.ping)
+        .catch((err) =>
+            console.error(
+                'Cannot connect to Backtrace in the main process.\n\n',
+                'Make sure to initialize @backtrace-labs/electron in the main process first.\n\n',
+                err,
+            ),
+        );
+
+    builder
+        .useRequestHandler(new IpcRequestHandler(ipcRpc))
+        .useReportSubmission(new IpcReportSubmission(ipcRpc, ipcTransport))
+        .useBreadcrumbsStorage(new IpcBreadcrumbsStorage(ipcTransport))
+        .useSummedMetricsQueue(new IpcSummedMetricsQueue(ipcTransport, ipcRpc))
+        .useUniqueMetricsQueue(new StubMetricsQueue());
+
+    if (options?.synchronous) {
+        const syncData = ipcRpc.invokeSync<SyncData>(IpcEvents.sync);
+        console.log(syncData);
+        builder.useSessionProvider(new ConstSessionProvider(syncData.sessionId));
+    } else {
+        builder.useSessionProvider(new IpcAsyncSessionProvider(ipcTransport));
```

Comment:
is there any reason anyone should prefer to have it instead of sync? 

The nature of programming is async all the way, however, it doesn't work well for error reporting solutions, while we're waiting on something, an error might happen. Can we stick to the sync session provider and introduce the async interface once we recognize we have to do something asynchronously? There is no real benefit other than "async all the way all the time without any good reason".

Reply:
The reason is that if you misconfigure something, there's no real way to tell the user "you screwed up" with synchronous code. It will just hang with a white screen.

I believe this may be counterintuitive, and may lead to the user removing Backtrace from his app and then seeing that removing fixed the white screen issue.

Reply:
so maybe we should always stick to the async one and update it in the future? Tbh if we update it as soon as possible, I believe it won't be a problem